tests/integration/plugins/test_uv.py

```diff
@@ -0,0 +1,272 @@
+
+    with lf.action_executor() as ctx:
+        ctx.execute(actions)
+
+    python_link = Path(lf.project_info.prime_dir, "bin", "python")
+    assert python_link.is_symlink()
+    assert python_link.resolve() == Path(sys.executable).resolve()
+
+
+@pytest.mark.parametrize("remove_symlinks", [(True), (False)])
+def test_uv_plugin_no_system_interpreter(
+    new_dir, partitions, uv_parts_simple, remove_symlinks: bool  # noqa: FBT001
+):
+    """Check that the build fails if a payload interpreter is needed but not found."""
+
+    class MyUvPlugin(craft_parts.plugins.plugins.UvPlugin):
+        @override
+        def _get_system_python_interpreter(self) -> str | None:
+            return None
+
+        @override
+        def _should_remove_symlinks(self) -> bool:
+            # Parametrize this to make sure that the build fails even if the
+            # venv symlinks will be removed.
+            return remove_symlinks
+
+    plugins.register({"uv": MyUvPlugin})
+
+    lf = LifecycleManager(
+        uv_parts_simple(),
+        application_name="test_uv",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lf.plan(Step.PRIME)
+
+    with lf.action_executor() as ctx, pytest.raises(errors.PluginBuildError):
+        ctx.execute(actions)
+
+
+def test_uv_plugin_remove_symlinks(new_dir, partitions, uv_parts_simple):
+    """Override symlink removal."""
+
+    class MyUvPlugin(craft_parts.plugins.plugins.UvPlugin):
+        @override
+        def _should_remove_symlinks(self) -> bool:
+            return True
+
+    plugins.register({"uv": MyUvPlugin})
+
+    lf = LifecycleManager(
+        uv_parts_simple(),
+        application_name="test_uv",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lf.plan(Step.PRIME)
+
+    with lf.action_executor() as ctx:
+        ctx.execute(actions)
+
+    python_link = Path(lf.project_info.prime_dir, "bin", "python")
+    assert python_link.exists() is False
+
+
+@pytest.fixture
+def uv_parts_complex() -> Callable[[str, str], dict[str, Any]]:
+    """A part whose override-build copies the system's Python interpreter
+    into the payload"""
+
+    def _inner(
+        payload_python: str,
+        real_python: str,
+        *,
+        source: Path = Path(__file__).parent / "test_uv",
+    ) -> dict[str, Any]:
+        parts_yaml = textwrap.dedent(
+            f"""\
+            parts:
+              foo:
+                plugin: uv
+                source: {source}
+                override-build: |
+                  # Put a binary called "{payload_python}" in the payload
+                  mkdir -p ${{CRAFT_PART_INSTALL}}/usr/bin
+                  cp {real_python} ${{CRAFT_PART_INSTALL}}/usr/bin/{payload_python}
+                  craftctl default
+            """
+        )
+
+        return cast(dict[str, Any], yaml.safe_load(parts_yaml))
+
+    return _inner
+
+
+def test_find_payload_python_bad_version(new_dir, partitions, uv_parts_complex):
```

Comment:
I'm undecided on whether *every* python plugin subclass needs to repeat these tests
since it's already here I guess might as well leave it

Reply:
Probably not, since it's really 3 tests for the `BasePythonPlugin`.

Reply:
I've created an issue to fix this next pulse: https://github.com/canonical/craft-parts/issues/935